concordium-smart-contract-testing/src/impls.rs

```diff
@@ -81,33 +90,188 @@ impl ChainParameters {
     }
 }
 
+impl ChainBuilder {
+    /// Create a new [`ChainBuilder`] for constructing the [`Chain`].
+    ///
+    /// Can also be created via the [`Chain::builder`] method.
+    ///
+    /// # Example
+    /// ```
+    /// # use concordium_smart_contract_testing::*;
+    ///
+    /// let chain = ChainBuilder::new()
+    ///             // Use zero or more builder methods, for example:
+    ///             .micro_ccd_per_euro(ExchangeRate::new_unchecked(50000, 1))
+    ///             .block_time(Timestamp::from_timestamp_millis(123))
+    ///             // Then build:
+    ///             .build()
+    ///             .unwrap();
+    /// ```
+    pub fn new() -> Self {
+        Self {
+            external_node_endpoint: None,
+            micro_ccd_per_euro:     None,
+            euro_per_energy:        None,
+            block_time:             None,
+        }
+    }
+
+    /// Configure a connection to an external Concordium node.
+    ///
+    /// The connection can be used for getting the current exchange rates
+    /// between CCD, Euro and Energy.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use concordium_smart_contract_testing::*;
+    /// let chain = Chain::builder()
+    ///     .external_node_connection("http://node.testnet.concordium.com:20000")
+    ///     .build()
+    ///     .unwrap();
+    /// ```
+    pub fn external_node_connection(mut self, endpoint: &str) -> Self {
```

Comment:
Why is this not an `Endpoint` or `Into<Endpoint>`? That is what we always use in Rust SDK examples.

I don't like using strings. In particular it means this library now has to have "malformed URL" error, which it does not need to have.

Reply:
Good point.